--- origsrc/xchat-2.8.8/src/fe-gtk/mmx_cmod.S	2009-08-16 04:40:18.000000000 -0500
+++ src/xchat-2.8.8/src/fe-gtk/mmx_cmod.S	2010-07-21 15:26:16.089045900 -0500
@@ -35,7 +35,7 @@
 #define gm	28(%ebp)
 #define bm	32(%ebp)
 
-#ifdef UNDERSCORE_SYMBOLS /* need this to link with msvc */
+#if defined(_WIN32) || defined(__CYGWIN__)
 #define SHADE_XIMAGE_15 _shade_ximage_15_mmx
 #define SHADE_XIMAGE_16 _shade_ximage_16_mmx
 #define SHADE_XIMAGE_32 _shade_ximage_32_mmx
@@ -524,7 +524,7 @@ HAVE_MMX:
 	popl	%ebx
 	ret
 
-#if defined(__GNUC__) && !defined(_WIN32)
+#if defined(__GNUC__) && !defined(_WIN32) && !defined(__CYGWIN__)
 .section .note.GNU-stack, "", @progbits
 .previous
 #endif
